### PR TITLE
Mempool default

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: lib/libutils/ext/include/mempool.h [hunk 1]
```diff
@@ -21,6 +21,15 @@ struct mempool;
 
 #define MEMPOOL_ALIGN	__alignof__(long)
 
+#if defined(__KERNEL__)
+/*
+ * System wide memory pool for temporary memory allocation, all allocations
+ * must be freed before allocating function returns in order to avoid
+ * dead-lock.
+ */
+extern struct mempool *mempool_default;
```
Comment (jforissier) [inline lib/libutils/ext/include/mempool.h]: "default" does not describe the purpose nor the constraint on the usage of this pool. How about `mempool_transient` or `mempool_temp` maybe?

File: core/lib/libtomcrypt/mpa_desc.c [hunk 2]
```diff
@@ -60,6 +60,7 @@ void init_mp_tomcrypt(void)
 	if (!mem.pool)
 		panic();
 	external_mem_pool = &mem;
+	mempool_default = mem.pool;
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/lib/libtomcrypt/mpa_desc.c]: I'll add some asserts

File: lib/libutils/ext/include/mempool.h (hunk 1 above)
Comment (jenswi-linaro) [reply lib/libutils/ext/include/mempool.h]: All mempool allocations are temporary, that part of the deal with this interface. Perhaps I should try to clarify that in the description of the interface instead.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jenswi-linaro** [comment]
Ping?

**Comment by jforissier** [comment]
Now I get it, thanks :)

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied

**Pull request merged**